addon-resizer/main.go

```diff
@@ -0,0 +1,175 @@
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const noValue = "MISSING"
+
+var (
+	// Flags to define the resource requirements.
+	baseCPU              = flag.String("cpu", noValue, "The base CPU resource requirement.")
+	cpuPerNode           = flag.String("extra-cpu", "0", "The amount of CPU to add per node.")
+	baseMemory           = flag.String("memory", noValue, "The base memory resource requirement.")
+	memoryPerNode        = flag.String("extra-memory", "0Mi", "The amount of memory to add per node.")
+	baseStorage          = flag.String("storage", noValue, "The base storage resource requirement.")
+	storagePerNode       = flag.String("extra-storage", "0Gi", "The amount of storage to add per node.")
+	recommendationOffset = flag.Int("recommendation-offset", 10, "A number from range 0-100. When the dependent's resources are rewritten, they are set to the closer end of the range defined by this percentage threshold.")
+	acceptanceOffset     = flag.Int("acceptance-offset", 20, "A number from range 0-100. The dependent's resources are rewritten when they deviate from expected by a percentage that is higher than this threshold. Can't be lower than recommendation-offset.")
+	// Flags to identify the container to nanny.
+	podNamespace  = flag.String("namespace", os.Getenv("MY_POD_NAMESPACE"), "The namespace of the ward. This defaults to the nanny pod's own namespace.")
+	deployment    = flag.String("deployment", "", "The name of the deployment being monitored. This is required.")
+	podName       = flag.String("pod", os.Getenv("MY_POD_NAME"), "The name of the pod to watch. This defaults to the nanny's own pod.")
+	containerName = flag.String("container", "pod-nanny", "The name of the container to watch. This defaults to the nanny itself.")
+	// Flags to control runtime behavior.
+	pollPeriodMillis = flag.Int("poll-period", 10000, "The time, in milliseconds, to poll the dependent container.")
+)
+
+func checkPercentageFlagBounds(flagName string, flagValue int) {
+	if flagValue < 0 || flagValue > 100 {
+		log.Fatalf("%s flag must be between 0 and 100 inclusively, was %d.", flagName, flagValue)
+	}
+}
+
+// GetClient returns a k8s clientset to the request from inside of cluster
+func GetClient() kubernetes.Interface {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Can not get kubernetes config: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Can not create kubernetes client: %v", err)
+	}
+
+	return clientset
+}
+
+func buildOutOfClusterConfig() (*rest.Config, error) {
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	if kubeconfigPath == "" {
+		kubeconfigPath = os.Getenv("HOME") + "/.kube/config"
+	}
+	return clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+}
+
+// GetClientOutOfCluster returns a k8s clientset to the request from outside of cluster
+func GetClientOutOfCluster() kubernetes.Interface {
```

Comment:
nit: GetClientOutOfClusterOrDie() (here and below)

Reply:
GetClientOutOfCluster() has a fallback to GetClient(). I'm changing the latter one to "OrDie()"